plugins/newsdownloader.koplugin/main.lua

```diff
@@ -315,4 +316,16 @@ function NewsDownloader:setCustomDownloadDirectory()
     }:chooseDir()
 end
 
+function NewsDownloader:onCloseDocument()
+    local document_full_path = self.ui.document.file
```

Comment:
You *could* do this if you actually used it two lines down, though it seems redundant either way. But not as is.

Reply:
I suppose I should explain this. What happens is.

1. Table lookup in self for ui.
2. Table lookup in ui for document.
3. Table lookup in document for file.

Therefore if you want to use it multiple times, you store it in a local variable to prevent this inefficiency. But there doesn't seem to be a point to it in this case.

Reply:
Thanks

Reply:
After changes that var is used again twice. No need to change here